src/test/java/seedu/address/commons/util/StringUtilTest.java

```diff
@@ -219,4 +220,26 @@ public void getDetails_nullGiven_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> StringUtil.getDetails(null));
     }
 
+    //---------------- Tests for removeExtraWhitespace --------------------------------------
+    /*
+     * Equivalence Partitions:
+     * - null
+     * - empty string
+     * - strings with no extra spaces in between words
+     * - strings with extra spaces in between words
+     * - strings with other whitespace characters in between words
+     */
+
+    @Test
+    public void removeExtraWhitespace_nullGiven_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> StringUtil.removeExtraWhitespace(null));
+    }
+
+    @Test
+    public void removeExtraWhitespace_validInputs_correctResult() {
+        assertEquals(StringUtil.removeExtraWhitespace(""), "");
+        assertEquals(StringUtil.removeExtraWhitespace("words words more words"), "words words more words");
+        assertEquals(StringUtil.removeExtraWhitespace("words    more    words       -flag"), "words more words -flag");
+        assertEquals(StringUtil.removeExtraWhitespace("words \t \n words"), "words words");
```

Comment:
I removed extra whitespace characters in order to accommodate `ls(lots of spaces)-contacts`. all commands will now have extra whitespace characters removed, such that two words are always separated by a single space character

not sure if it's going to create problems